src/app/home/components/Card/Status/CardStatus.tsx

```diff
@@ -6,28 +6,16 @@ import { Grid, GridItem } from '@patternfly/react-core';
 
 const CardStatusComponent = ({ type, dataList, ...props }) => {
   let successList = [];
-  const failureList = [];
   if (type === 'repositories') {
-    // hasReadyCondition = plan.MigPlan.status.conditions.filter(c => c.type === 'Ready').length > 0;
-    // hasErrorCondition =
-    //   plan.MigPlan.status.conditions.filter(c => c.category === 'Critical').length > 0;
-
-    successList = dataList.filter(item => {
-      if (item.MigStorage.status) {
-        return item.MigStorage.status.conditions.filter(c => c.type === 'Ready');
-      } else {
-        failureList.push(item);
-      }
-    });
+    successList = dataList.filter(item =>
+      item.MigStorage.status && item.MigStorage.status.conditions.filter(c => c.type === 'Ready').length !== 0
+      );
   } else if (type === 'clusters') {
-    successList = dataList.filter(item => {
-      if (item.MigCluster.status) {
-        return item.MigCluster.status.conditions.filter(c => c.type === 'Ready');
-      } else {
-        failureList.push(item);
-      }
-    });
+    successList = dataList.filter(item =>
+      item.MigCluster.status && item.MigCluster.status.conditions.filter(c => c.type === 'Ready').length !== 0
+    );
   }
+  const failureList = dataList.filter(item => successList.indexOf(item) < 0);
```

Comment:
Although this happens to work due to some specific circumstances, it's can be a very error prone way to check objects for semantic equality (`indexOf` uses strict comparison, `===`). It will in fact return (this example does). You have to be extremely careful when you are trying to see if two objects are "equal" to one another, because "equal" can mean very different things depending on the context:

```
const foo = {name: 'hello', deeperThing: { age: 10 }};
const bar = {name: 'hello', deeperThing: { age: 10 }};

console.log(foo === bar)
===
Output: false
```

A better check here would be to retrieve the set of objects that are **not** found in the successful set, based on their names:

```diff
diff --git a/src/app/home/components/Card/Status/CardStatus.tsx b/src/app/home/components/Card/Status/CardStatus.tsx
index 4e7e350..f2cc7d9 100644
--- a/src/app/home/components/Card/Status/CardStatus.tsx
+++ b/src/app/home/components/Card/Status/CardStatus.tsx
@@ -6,16 +6,22 @@ import { Grid, GridItem } from '@patternfly/react-core';
 
 const CardStatusComponent = ({ type, dataList, ...props }) => {
   let successList = [];
+  let failureList = [];
+
   if (type === 'repositories') {
     successList = dataList.filter(item =>
       item.MigStorage.status && item.MigStorage.status.conditions.filter(c => c.type === 'Ready').length !== 0
       );
+    const successfulNames = successList.map(item => item.MigStorage.metadata.name);
+    failureList = dataList.filter(item => !successfulNames.includes(item.MigStorage.metadata.name));
   } else if (type === 'clusters') {
     successList = dataList.filter(item =>
       item.MigCluster.status && item.MigCluster.status.conditions.filter(c => c.type === 'Ready').length !== 0
     );
+    const successfulNames = successList.map(item => item.MigCluster.metadata.name);
+    failureList = dataList.filter(item => !successfulNames.includes(item.MigCluster.metadata.name));
   }
-  const failureList = dataList.filter(item => successList.indexOf(item) < 0);
+
   return (
     <React.Fragment>
       <Grid>
```